Fix try_compile() not using correct linker.

Manually passes CMAKE_EXE_LINKER_FLAGS to try_compile() in order to use Gold linker.
For some reason enabling CMake policy CMP0056 did not work here.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7f8a5bd..91efde8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -215,7 +215,7 @@ function(test_cxx CXX_RUNTIME_NAME IS_STDLIB)
 		message(STATUS "Testing ${CXX_RUNTIME_LIB} as the C++ runtime library")
 		if (ANDROID)
 			# pass on Android toolchain flags
-			set(CXX_RUNTIME_FLAGS "-DANDROID_NDK=${ANDROID_NDK}" "-DANDROID_ABI=${ANDROID_ABI}" "-DANDROID_ARM_MODE=${ANDROID_ARM_MODE}" "-DANDROID_ARM_NEON=${ANDROID_ARM_NEON}" "-DANDROID_LD=${ANDROID_LD}" "-DANDROID_PLATFORM=${ANDROID_PLATFORM}" "-DANDROID_STL=${ANDROID_STL}")
+			set(CXX_RUNTIME_FLAGS "-DCMAKE_EXE_LINKER_FLAGS=${CMAKE_EXE_LINKER_FLAGS}" "-DANDROID_NDK=${ANDROID_NDK}" "-DANDROID_ABI=${ANDROID_ABI}" "-DANDROID_ARM_MODE=${ANDROID_ARM_MODE}" "-DANDROID_ARM_NEON=${ANDROID_ARM_NEON}" "-DANDROID_LD=${ANDROID_LD}" "-DANDROID_PLATFORM=${ANDROID_PLATFORM}" "-DANDROID_STL=${ANDROID_STL}")
 		endif()
 		try_compile(USERUNTIME 
 			"${CMAKE_BINARY_DIR}/CMake"
